src/configurePipeline/clients/azureDevOpsClient.ts

```diff
@@ -0,0 +1,241 @@
+                        "tenantid": inputs.authDetails.tenantId
+                    },
+                    "scheme": "ServicePrincipal"
+                },
+                "data": {
+                    "appObjectId": "",
+                    "azureSpnPermissions": "",
+                    "azureSpnRoleAssignmentId": "",
+                    "creationMode": "Automatic",
+                    "environment": "AzureCloud",
+                    "scopeLevel": "Subscription",
+                    "spnObjectId": "",
+                    "subscriptionId": inputs.subscriptionId,
+                    "subscriptionName": inputs.subscriptionId
+                },
+                "description": "",
+                "groupScopeId": null,
+                "name": endpointName,
+                "operationStatus": null,
+                "readersGroup": null,
+                "type": "azurerm",
+                "url": "https://management.azure.com/"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async getEndpointStatus(endpointId: string, organizationName: string, projectName: string, ): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: "https://" + organizationName + ".visualstudio.com/" + projectName + "/_apis/serviceendpoint/endpoints/" + endpointId,
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json;api-version=5.1-preview.2;excludeUrls=true"
+            },
+            method: "Get",
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async authorizeEndpointForAllPipelines(endpointId: string, endpointName: string, organizationName: string, projectName: string): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: "https://" + organizationName + ".visualstudio.com/" + projectName + "/_apis/pipelines/pipelinePermissions/endpoint/" + endpointId,
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json;api-version=5.1-preview.1;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true"
+            },
+            method: "PATCH",
+            body: {
+                "allPipelines": {
+                    "authorized": true,
+                    "authorizedBy": null,
+                    "authorizedOn": null
+                },
+                "pipelines": null,
+                "resource": {
+                    "id": endpointId,
+                    "type": "endpoint"
+                }
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async createAndRunPipeline(pipelineConfiguration: WizardInputs): Promise<any> {
```

Comment:
this is not a documented api. i would rather call the rest api to create build pipeline